cve_bin_tool/cli.py

```diff
@@ -223,42 +255,56 @@ def main(argv=None):
             )
         )
 
-    with CVEScanner() as cve_scanner:
-        scanner = VersionScanner(cve_scanner, args.extract, score=score)
-        scanner.remove_skiplist(skips)
-        LOGGER.info(scanner.print_checkers())
-        scanner.recursive_scan(args.directory)
+    with CVEScanner(score=score) as cve_scanner:
+        total_files = 0
+        parsed_data = {}
+        if args.input_file:
+            input_engine = InputEngine(args.input_file)
+            parsed_data = input_engine.parse_input()
+            if not args.directory:
+                for params, remarks in parsed_data.items():
+                    cve_scanner.get_cves(*params, remarks)
+        if args.directory:
+            version_scanner = VersionScanner(should_extract=args.extract)
+            version_scanner.remove_skiplist(skips)
+            LOGGER.info(version_scanner.print_checkers())
+            for params in version_scanner.recursive_scan(args.directory):
+                if params:
+                    LOGGER.debug(params)
+                    remarks = parsed_data.get(params, Remarks.NewFound)
+                    cve_scanner.get_cves(*params, remarks)
+            total_files = version_scanner.total_scanned_files
 
         LOGGER.info("")
         LOGGER.info("Overall CVE summary: ")
         LOGGER.info(
-            f"There are {scanner.files_with_cve} files with known CVEs detected"
+            f"There are {cve_scanner.products_with_cve} products with known CVEs detected"
```

Comment:
just a better name because files_with_cves won't make much sense while using it as csv2cve

Reply:
files_with_cves is the actual number of files scanned that contained a product with CVEs

Reply:
Good thinking, though I wonder if products will be confusing for people scanning directories, as they may think of a "product" as a .deb or somesuch and we have a different definition.  it's tempting to just say "things" instead of files or products, but I guess that doesn't sound quite as formal.